components/org.wso2.carbon.identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java

```diff
@@ -0,0 +1,358 @@
+import org.wso2.carbon.identity.oauth.uma.resource.endpoint.dto.ErrorDTO;
+import org.wso2.carbon.identity.oauth.uma.resource.endpoint.dto.ReadResourceDTO;
+import org.wso2.carbon.identity.oauth.uma.resource.endpoint.dto.ResourceDetailsDTO;
+import org.wso2.carbon.identity.oauth.uma.resource.endpoint.dto.UpdateResourceDTO;
+
+import org.wso2.carbon.identity.oauth.uma.resource.endpoint.exceptions.ResourceEndpointException;
+import org.wso2.carbon.identity.oauth.uma.resource.endpoint.util.ResourceUtils;
+
+import org.wso2.carbon.identity.oauth.uma.resource.service.ResourceConstants;
+import org.wso2.carbon.identity.oauth.uma.resource.service.exceptions.UMAClientException;
+import org.wso2.carbon.identity.oauth.uma.resource.service.exceptions.UMAException;
+import org.wso2.carbon.identity.oauth.uma.resource.service.exceptions.UMAServiceException;
+import org.wso2.carbon.identity.oauth.uma.resource.service.model.Resource;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.ws.rs.core.Response;
+
+
+/**
+ * ResourceRegistrationApiServiceImpl is used to handling resource management.
+ */
+public class ResourceRegistrationApiServiceImpl extends ResourceRegistrationApiService {
+
+    private static final Log log = LogFactory.getLog(ResourceRegistrationApiServiceImpl.class);
+    private final String patScope = "uma_protection";
+    private final String authContext = "auth-context";
+    private final String oauth2AllowedScopes = "oauth2-allowed-scopes";
+
+    /**
+     * Register a resource with resource details
+     *
+     * @param requestedResource details of the resource to be registered
+     * @return Response with the status of the registration
+     */
+    @Override
+    public Response registerResource(ResourceDetailsDTO requestedResource, MessageContext context) {
+
+        Response response = null;
+
+        if (validateTokenScope(context)) {
+            if (requestedResource.getResource_scopes() == null || requestedResource.getResource_scopes().isEmpty()) {
+                log.error("Request body cannot be empty and should consist with scopes.");
+                return Response.status(Response.Status.BAD_REQUEST).build();
+            }
+            try {
+                Resource registerResource = ResourceUtils.getResourceService()
+                        .registerResource(ResourceUtils.getResource(requestedResource),
+                                getResourceOwner(context), getTenantIdFromCarbonContext(), getConsumerKey(context));
+                CreateResourceDTO createResourceDTO = ResourceUtils.createResponse(registerResource);
+                return Response.status(Response.Status.CREATED).entity(createResourceDTO).build();
+
+            } catch (UMAServiceException e) {
+                handleErrorResponse(e, true, log);
+            } catch (UMAClientException e) {
+                handleErrorResponse(e, true, log);
```

Comment:
log is not used, so why passing log? Shall we put a proper error message here and change the method